src/ORM/Association/Loader/SelectLoader.php

```diff
@@ -168,7 +168,7 @@ protected function _buildQuery(array $options): SelectQuery
         assert($query instanceof SelectQuery);
         if (isset($options['finder'])) {
             [$finderName, $opts] = $this->_extractFinder($options['finder']);
-            $query = $query->find($finderName, $opts);
+            $query = $query->find($finderName, ...$opts);
```

Comment:
This seems like it is going to give a big upgrade bump for userland code. I think we should change the implementation of `find()` so that we can accept both the old form of `find($finderName, $opts)` as that will be pervasive in userland code.

Reply:
Hmm so perhaps change the signature to `find($finder, array $options, mixed ...$args)`?

Reply:
Or  `find($finder, mixed ...$args)` and then in `find()` we need to handle detecting if we got a single array of options, or named parameters. I think the formats are different enough that we can do it. If we get parameters that don't strictly match one format or the other we could raise errors.

Reply:
All `find()` call are eventually invoked through `Table::invokeFinder()` which takes care of BC usage, so I don't think there's any issue here. Please add more test cases to ensure BC compatibility if existing tests have been updated as per new signature.